database/engine/rrdengineapi.c

```diff
@@ -397,7 +397,7 @@ unsigned rrdeng_variable_step_boundaries(RRDSET *st, time_t start_time, time_t e
     page_info_array = NULL;
 
     rrdset_rdlock(st);
-    for(rd_iter = st->dimensions, rd = NULL, min_time = (usec_t)-1 ; rd_iter ; rd_iter = rd_iter->next) {
+    for(rd_iter = temp_rd?temp_rd:st->dimensions, rd = NULL, min_time = (usec_t)-1 ; rd_iter ; rd_iter = rd_iter->next) {
```

Comment:
We already talked about this on Slack, but I will add it here for we do not forget if a problem happens. We are locking the chart here, but this function when is called from `rrd2rrdr` is not locking the host. So it is called without host lock.

Reply:
While in general we can discuss this, I don't think it is a regression of this PR in particular. If this is a problem, it was always a problem.

Reply:
I did not add additional locks (only as required by the rrdset_foreach_read / rrddim_foreach_read)